devtools/running_packages.rst

```diff
@@ -114,6 +113,214 @@ with:
+For instance, if the application depends on boost, we may not know that it also requires many other 3rt-party libraries, 
+such as 
+`zlib <https://zlib.net/>`_, 
+`bzip2 <https://sourceware.org/bzip2/>`_, 
+`lzma <https://tukaani.org/xz/>`_, 
+`zstd <https://facebook.github.io/zstd/>`_, 
+`iconv <https://www.gnu.org/software/libiconv/>`_, etc. 
+
+.. code-block:: bash
+
+    $ conan install . -g deploy
+
+This helps to collect all the dependencies into a single place, moving them out of the conan cache directory.
+
+.. _using_json_generator:
+
+Using json generator
+--------------------
+
+A more advanced approach is to use the :ref:`json generator <json_generator>`:
+file containing all required information about dependencies:
+
+.. code-block:: bash
+
+    $ conan install . -g json
+
+The `conanbuildinfo.json` file produced is fully machine-readable and could be used by scripts to prepare the distribution.
+
+.. code-block:: python
+
+        import os
+        import json
+
+        data = json.load(open("conanbuildinfo.json"))
+
+        dep_lib_dirs = dict()
+        dep_bin_dirs = dict()
+
+        for dep in data["dependencies"]:
+            root = dep["rootpath"]
+            lib_paths = dep["lib_paths"]
+            bin_paths = dep["bin_paths"]
+
+            dep_lib_dirs = dict()
+            dep_bin_dirs = dict()
+
+            for lib_path in lib_paths:
+                if os.listdir(lib_path):
+                    lib_dir = os.path.relpath(lib_path, root)
+                    dep_lib_dirs[lib_path] = lib_dir
+            for bin_path in bin_paths:
+                if os.listdir(bin_path):
+                    bin_dir = os.path.relpath(bin_path, root)
+                    dep_bin_dirs[bin_path] = bin_dir
+
+The advantage over the ``deploy`` generator is fine-grained control: here we copy only binaries and libraries.
+It's also could be easily modified to apply some sort of filtering (e.g. to copy only shared libraries, 
+and omit any static libraries or auxiliary files such as pkg-config .pc files).
+The extracted information may also be used to generate a simple startup script (as described below):
+
+.. code-block:: python
+
+    executable = "MyApp"  # just an example
+    varname = "$APPDIR"
+
+    def _format_dirs(dirs):
+        return ":".join(["%s/%s" % (varname, d) for d in dirs])
+
+    path = _format_dirs(bin_dirs.values())
+    ld_library_path = _format_dirs(bin_dirs.values())
+    exe = varname + "/" + executable
+
+    content = """#!/usr/bin/env bash
+    set -ex
+    export PATH=$PATH:{path}
+    export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:{ld_library_path}
+    pushd $(dirname {exe})
+    $(basename {exe})
+    popd
+    """.format(path=path,
+           ld_library_path=ld_library_path,
+           exe=exe)
+
+.. note::
+
+    The full example might be found on `GitHub <https://github.com/SSE4/conan-deploy-tool>`_.
+
+.. _deployment_challenges:
```

Comment:
IMO this part does not fit very well in here. But I am not sure where we should put this

Reply:
I am open for suggestions